### DISCUSSION TITLE
这个能玩的游戏竟然是用着色器语言写出来的

### DISCUSSION THREAD

**Discussion by mantou132** [just-talks/tech-talks#43] [3.分享创造]
https://www.shadertoy.com/view/XltGDr

那些输入是平台内置的，图像都是生成的。

https://www.shadertoy.com/view/XtlSD7
这个项目的声音都是生成的。

当然，这里面任何一个项目都能震撼我一整年

<img width="445" alt="Screenshot 2023-04-01 at 01 12 24" src="https://user-images.githubusercontent.com/3841872/229186206-d9510121-2ae1-46fb-8a39-2b5504e776dc.png">

**Comment by adoyle-h** [comment]
我想到了用纯 CSS 写的游戏。比如这个[冒险游戏](https://codepen.io/jcoulterdesign/pen/NOMeEb)，和这个[小游戏](https://codepen.io/ivorjetski/pen/OJXbvdL)。也都很震撼。
用着色器语言或者 CSS 写游戏都是炫技之作，并不适合做工程。因为代码复用率低，修改和调试也都很麻烦。
